cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -283,6 +293,66 @@ public Future<Void> reconcile(KafkaStatus kafkaStatus, Clock clock)    {
+
+        Set<Integer> newAddedBrokers = new LinkedHashSet<>(kafka.addedBrokerNodes());
+        if (kafkaAutoRebalanceStatus == null) {
+            // no already stored added nodes, adding only the new ones from current reconciliation
+            if (!newAddedBrokers.isEmpty()) {
+                kafkaStatus.setAutoRebalance(
+                        new KafkaAutoRebalanceStatusBuilder()
+                                .withModes(
+                                        new KafkaAutoRebalanceModeBrokersBuilder()
+                                                .withMode(KafkaRebalanceMode.ADD_BROKERS)
+                                                .withBrokers(kafka.addedBrokerNodes().stream().toList())
+                                                .build()
+                                )
+                                .build());
+            }
+            return Future.succeededFuture();
+        }
+
+        KafkaAutoRebalanceStatusBuilder builder = new KafkaAutoRebalanceStatusBuilder(kafkaAutoRebalanceStatus);
+        // no already stored added nodes, adding only the new ones from current reconciliation
+        if (kafkaAutoRebalanceStatus.getModes() == null) {
+            if (!newAddedBrokers.isEmpty()) {
+                builder.withModes(
+                        new KafkaAutoRebalanceModeBrokersBuilder()
+                                .withMode(KafkaRebalanceMode.ADD_BROKERS)
+                                .withBrokers(kafka.addedBrokerNodes().stream().toList())
+                                .build()
+                );
+            }
+            kafkaStatus.setAutoRebalance(builder.build());
+            return Future.succeededFuture();
+        }
+        // going to create a consistent list between:
+        // the added nodes already stored in the status
+        // the newly added nodes within the current reconciliation
+
+        // Merge the already stored brokers with new ones if the mode exists, or create a new mode
+        Optional<KafkaAutoRebalanceModeBrokers> existingAddBrokersMode = kafkaAutoRebalanceStatus.getModes().stream()
+                .filter(mode -> mode.getMode().equals(KafkaRebalanceMode.ADD_BROKERS))
+                .findFirst();
+
+        if (existingAddBrokersMode.isPresent()) {
+            newAddedBrokers.addAll(existingAddBrokersMode.get().getBrokers());
+            builder.editMatchingMode(m -> m.getMode().equals(KafkaRebalanceMode.ADD_BROKERS))
+                        .withBrokers(newAddedBrokers.stream().toList())
+                    .endMode();
+        } else if (!newAddedBrokers.isEmpty()) {
+            builder.addNewMode()
+                        .withMode(KafkaRebalanceMode.ADD_BROKERS)
+                        .withBrokers(newAddedBrokers.stream().toList())
+                    .endMode();
+        }
+        kafkaStatus.setAutoRebalance(builder.build());
+        return Future.succeededFuture();
```

Comment:
I wonder if this could be some nice static method somewhere in some utils class for easier testing and better code readability. The code looks quite complex, so being able to give it a good coverage would seem handy.

Reply:
I have factored out the method into the `KafkaRebalanceUtils` class. If you think that's fine I can add some tests for it. Let me know.

Reply:
FYI I have already added some unit test in the `KafkaRebalanceUtilsTest` class.